python/gink/crud.py

```diff
@@ -0,0 +1,149 @@
+"""
+The WSGI API application for gink operations.
+
+To run this application, pass --wsgi gink.crud.app when starting gink.
+
+Set auth key with env AUTH_TOKEN.
+
+"""
+
+from json import loads, dumps
+from os import environ
+
+from .impl.directory import Directory
+
+
+class Crud():
+    """
+    WSGI application to GET, PUT, and DELETE data in a Gink database.
+
+    Example for PUT:
+
+    curl -X PUT http://localhost:8099/key -d 3 -H "Content-Type: application/json"
+
+    # Sets "key" in root directory to 3
+
+    Example for GET:
+
+    curl -X GET http://localhost:8099/key
+
+    # Returns 3
+
+    """
+    def __init__(self):
+        self.root = Directory(arche=True)
+        self.auth_token = environ.get("AUTH_TOKEN")
+
+    def __call__(self, environ, start_response):
+        # If auth token is present, expect token in the Authorization header.
+        if self.auth_token:
+            auth_header = environ.get('HTTP_AUTHORIZATION')
+            if not auth_header or self.auth_token not in auth_header:
```

Comment:
To allow for instances where authorization isn't necessary, I think it would be best to skip checking the header if no auth token has been passed in through the `os.environ`.

Reply:
Yeah actually you're right I misread it.